src/shared/lib/image.ts

```diff
@@ -100,3 +102,74 @@ export const compressFileImage: compressFileImage = async (file, options) => {
 
   return compressedFile;
 };
+
+interface imageState {
+  [key: string]: { isSuccess: boolean };
+}
+
+export const useImageState = (source?: string | string[]) => {
+  const [isLoading, setIsLoading] = useState(() => (source ? true : false));
+  const [imageState, setImageState] = useState<imageState>({});
+
+  const loadImage = async (source: string | string[]) => {
+    if (typeof source === "string") {
+      if (!imageState[source]) {
+        return;
+      }
+      setIsLoading(true);
+      const img = new Image();
+      img.src = source;
+      img.onload = () => {
+        setIsLoading(false);
+        setImageState({ [source]: { isSuccess: true } });
+      };
+      img.onerror = () => {
+        setIsLoading(false);
+        setImageState({ [source]: { isSuccess: false } });
+      };
+      return;
```

Comment:
가독성이 더 좋아지도록 띄어 쓰기를 하면 어떨까요?!
개인적으로 선언부 / 실행부 사이에 띄어쓰기하는 걸 선호합니다!

```suggestion
      if (!imageState[source]) {
        return;
      }
      
      setIsLoading(true);
      const img = new Image();
      
      img.src = source;
      img.onload = () => {
        setIsLoading(false);
        setImageState({ [source]: { isSuccess: true } });
      };
      img.onerror = () => {
        setIsLoading(false);
        setImageState({ [source]: { isSuccess: false } });
      };
      
      return;
```